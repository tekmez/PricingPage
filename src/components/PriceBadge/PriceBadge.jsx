import { Text, View } from 'react-native'
import styles from './PriceBadge.style'
const PriceBadge = ({ title }) => {
  return (
    <View style={styles.container}>
      <Text testID="price-title" style={styles.title}>
        {title}
      </Text>
    </View>
  )
}

export default PriceBadge
