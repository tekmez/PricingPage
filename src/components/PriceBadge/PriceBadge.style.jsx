import { StyleSheet } from 'react-native'
const styles = StyleSheet.create({
  container: {
    paddingVertical: 2,
    paddingHorizontal: 16,
    borderRadius: 16,
    backgroundColor: '#FF0000',
    width: 92,
    height: 20,
    position: 'absolute',
    top: -22,
    left: 245,
  },
  title: {
    fontSize: 12,
    fontFamily: 'Inter-Regular',
    color: '#FFFFFF',
    lineHeight: 16,
  },
})
export default styles
