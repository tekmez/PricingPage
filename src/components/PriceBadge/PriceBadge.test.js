import React from 'react'
import { render } from '@testing-library/react-native'
import PriceBadge from './PriceBadge'

describe('Price Badge', () => {
  it('should match snapshot', () => {
    const comp = render(<PriceBadge />).toJSON()
    expect(comp).toMatchSnapshot()
  })

  it('should render correctly', () => {
    const testTite = 'Price Badge'
    const comp = render(<PriceBadge title={testTite} />)
    const priceBadgeTitle = comp.getByTestId('price-title').children[0]
    expect(priceBadgeTitle).toBe(testTite)
  })
})
