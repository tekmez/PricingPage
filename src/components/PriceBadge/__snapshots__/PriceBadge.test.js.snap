// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price Badge should match snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#FF0000",
      "borderRadius": 16,
      "height": 20,
      "left": 245,
      "paddingHorizontal": 16,
      "paddingVertical": 2,
      "position": "absolute",
      "top": -22,
      "width": 92,
    }
  }
>
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontFamily": "Inter-Regular",
        "fontSize": 12,
        "lineHeight": 16,
      }
    }
    testID="price-title"
  />
</View>
`;
